test/CMakeLists.txt

```diff
@@ -85,6 +85,8 @@ if(BUILD_TESTS)
       Gather_InPlace.cpp
       Gather_ManagedMem.cpp
       Gather_OutOfPlace.cpp
+      #SendRecv
+      SendRecv_OutOfPlace.cpp
```

Comment:
SendRecv only works on out-of-place data.
This one is actually SendRecv_Pairwise.cpp because we are testing just pairwise connections.
In addition, the next test can be SendRecv_AllPairs.cpp which should test all send receives amongst all ranks in parallel (like an AllToAll)